src/sql/src/plan/statement/ddl.rs

```diff
@@ -3236,20 +3241,16 @@ pub fn plan_drop_database(
         if_exists,
     }: DropDatabaseStatement,
 ) -> Result<Plan, PlanError> {
-    let id = match scx.resolve_database(&name) {
-        Ok(database) => {
-            if restrict && database.has_schemas() {
-                sql_bail!(
-                    "database '{}' cannot be dropped with RESTRICT while it contains schemas",
-                    name,
-                );
-            }
-            Some(database.id())
+    let database = resolve_database(scx, &name, if_exists)?;
```

Comment:
Does this need an RBAC check?

Reply:
All of the drops and alters will need RBAC checks to make sure that the user is the owner or a superuser. That's going to be my next PR.